File: src/FetchMeditation/JFT.php
<?php

namespace FetchMeditation;

abstract class JFT
{
    protected JFTSettings $settings;

    public function __construct(JFTSettings $settings)
    {
        $this->settings = $settings;
    }

    abstract protected function fetch();

    abstract protected function getLanguage();

    public static function getInstance(JFTSettings $settings): JFT
    {
        return match ($settings->language) {
            JFTLanguage::English => new EnglishJFT($settings),
            JFTLanguage::French => new FrenchJFT($settings),
            JFTLanguage::German => new GermanJFT($settings),
            JFTLanguage::Italian => new ItalianJFT($settings),
            JFTLanguage::Japanese => new JapaneseJFT($settings),
            JFTLanguage::Portuguese => new PortugueseJFT($settings),
            JFTLanguage::Russian => new RussianJFT($settings),
            JFTLanguage::Spanish => new SpanishJFT($settings),
            JFTLanguage::Swedish => new SwedishJFT($settings),
        };
    }
}

File: py/JFT.py
from JFTLanguage import JFTLanguage
from GermanJFT import GermanJFT
from EnglishJFT import EnglishJFT


class JFT:
    def __init__(self, settings):
        self.settings = settings

    def fetch(self):
        pass

    def get_language(self):
        pass

    @staticmethod
    def get_instance(settings):
        return {
            JFTLanguage.English: EnglishJFT,
            JFTLanguage.German: GermanJFT,
        }[settings.language](settings)
